packages/features/ee/deployment/licensekey/CreateLicenseKeyForm.tsx

```diff
@@ -0,0 +1,270 @@
+import type { SessionContextValue } from "next-auth/react";
+import { useSession } from "next-auth/react";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc";
+import type { Ensure } from "@calcom/types/utils";
+import { showToast } from "@calcom/ui";
+import { Alert, Button, Form, Label, TextField, ToggleGroup } from "@calcom/ui";
+
+import { UserPermissionRole } from "../../../../prisma/enums";
+
+export const CreateANewLicenseKeyForm = () => {
+  const session = useSession();
+  if (session.data?.user.role !== "ADMIN") {
+    return null;
+  }
+  // @ts-expect-error session can't be null due to the early return
+  return <CreateANewLicenseKeyFormChild session={session} />;
+};
+
+enum BillingType {
+  PER_BOOKING = "PER_BOOKING",
+  PER_USER = "PER_USER",
+}
+
+enum BillingPeriod {
+  MONTHLY = "MONTHLY",
+  ANNUALLY = "ANNUALLY",
+}
+
+interface FormValues {
+  billingType: BillingType;
+  entityCount: number;
+  entityPrice: number;
+  billingPeriod: BillingPeriod;
+  overages: number;
+  billingEmail: string;
+}
+
+const CreateANewLicenseKeyFormChild = ({ session }: { session: Ensure<SessionContextValue, "data"> }) => {
+  const { t } = useLocale();
+  const [serverErrorMessage, setServerErrorMessage] = useState<string | null>(null);
+  const [stripeCheckoutUrl, setStripeCheckoutUrl] = useState<string | null>(null);
+  const isAdmin = session.data.user.role === UserPermissionRole.ADMIN;
+  const newLicenseKeyFormMethods = useForm<FormValues>({
+    defaultValues: {
+      billingType: BillingType.PER_BOOKING,
+      billingPeriod: BillingPeriod.MONTHLY,
+      entityCount: 500,
+      overages: 99, // $0.99
+      entityPrice: 50, // $0.5
+      billingEmail: undefined,
+    },
+  });
+
+  const mutation = trpc.viewer.admin.createSelfHostedLicense.useMutation({
+    onSuccess: async (values) => {
+      showToast(`Success: We have created a stripe payment URL for this billing email`, "success");
+      setStripeCheckoutUrl(values.stripeCheckoutUrl);
+    },
+    onError: async (err) => {
+      setServerErrorMessage(err.message);
+    },
+  });
+
+  const watchedBillingPeriod = newLicenseKeyFormMethods.watch("billingPeriod");
+  const watchedEntityCount = newLicenseKeyFormMethods.watch("entityCount");
+  const watchedEntityPrice = newLicenseKeyFormMethods.watch("entityPrice");
+
+  function calculateMonthlyPrice() {
+    const occurance = watchedBillingPeriod === "MONTHLY" ? 1 : 12;
```

Comment:
Nit: We should enforce American spelling

```suggestion
    const occurrence = watchedBillingPeriod === "MONTHLY" ? 1 : 12;
```